uaclient/contract.py

```diff
@@ -281,11 +321,25 @@ def _get_activity_info(self, machine_id: Optional[str] = None):
             if ent(self.cfg).user_facing_status()[0] == UserFacingStatus.ACTIVE
         ]
 
-        return {
-            "activityID": activity_id,
-            "activityToken": self.cfg.activity_token,
-            "resources": enabled_services,
-        }
+        uses_fips_java = False
+        if "fips" in enabled_services or "fips-updates" in enabled_services:
+            java_package_matches = apt.search_installed_packages(
```

Comment:
@orndorffgrant shouldn't we go the other way around ? Identify first which `openjdk-fips` package is installed and then just query for a specific version of `openjdk-jre-headless` package ? Or can we have more than one `openjdk-fips` package in the machine ?

Reply:
I don't know :)

It may be the case that all the -fips packages get installed as part of enable fips, in which case the way it is now is better. Or maybe only one will get installed manually.

After we have more concrete requirements, we can come back to this